src/core/thread/mesh_forwarder.cpp

```diff
@@ -1256,6 +1255,14 @@ void MeshForwarder::UpdateSendMessage(Error aFrameTxError, Mac::Address &aMacDes
         break;
 
     case Message::kSubTypeMleChildIdRequest:
+        if (mSendMessage->GetTxSuccess())
+        {
+            if (!Get<Mle::Mle>().IsRxOnWhenIdle())
+            {
+                Get<DataPollSender>().SetAttachMode(true);
+                Get<MeshForwarder>().SetRxOnWhenIdle(false);
+            }
+        }
```

Comment:
Can combine the two `if`s?
```suggestion
        if (mSendMessage->GetTxSuccess() && !Get<Mle::Mle>().IsRxOnWhenIdle())
        {
             Get<DataPollSender>().SetAttachMode(true);
             Get<MeshForwarder>().SetRxOnWhenIdle(false);
        }
        
```

Reply:
Sure, combined the ifs